### DISCUSSION TITLE
Character generator far less code

### DISCUSSION THREAD

**Discussion by hayesgr** [Z3ROCYPHER/pcg#15] [Show and tell]
I seen your project on a youtube code review.
I think you over complicated the generator.
I posted two versions the you can find both in the same file. Just comment one out.
The smaller of the two is only 20 lines of code if you remove the comments. Out of that 5 are arrays of data.
https://github.com/hayesgr/RandomString/blob/main/generate.c
Hope you find it helpful or educational.

**Comment by Z3ROCYPHER** [comment]
Thanks for your suggestion, I'm going to verify better later, right now I am about to change some things in the project.

**Reply by hayesgr** [reply]
Well good luck with the project. It will be there for you to check out when you get the chance. I added another commented out version at the bottom. It reduces the number of lines by 4 reducing them to 16 lines.

One suggestion. I would move any input checking into the UI section. That is more part of the HMI. If a developer wanted to use it for say a server he could set the length and types of values fed to it and never would need to worry about it being incorrect. It is only when dealing with people it is needed.
